stdlib/3/builtins.pyi

```diff
@@ -738,10 +738,10 @@ class list(MutableSequence[_T], Generic[_T]):
         def __class_getitem__(cls, item: Any) -> GenericAlias: ...
 
 class dict(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
-    # NOTE: Keyword arguments are special. If they are used, _KT must include
-    #       str, but we have no way of enforcing it here.
     @overload
-    def __init__(self, **kwargs: _VT) -> None: ...
+    def __init__(self: Dict[Any, Any]) -> None: ...
+    @overload
+    def __init__(self: Dict[str, _VT], **kwargs: _VT) -> None: ...
```

Comment:
Technically the key type could also be a union including str. It's not terribly likely to occur in practice, but the change may lead to false positives on code like:
```
x: Dict[Union[str, int], int] = dict(x=1, y=2)
x[1] = "y"
```
I'd favor ignoring this scenario until somebody comes up with a real-world example, since the scenarios where this change helps are a lot more likely to appear in practice.